tests/Schema/TableDiffTest.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Schema;
+
+use Doctrine\DBAL\Schema\Column;
+use Doctrine\DBAL\Schema\ForeignKeyConstraint;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Schema\TableDiff;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\Deprecations\PHPUnit\VerifyDeprecations;
+use PHPUnit\Framework\TestCase;
+
+class TableDiffTest extends TestCase
+{
+    use VerifyDeprecations;
+
+    public function testCreateWithInvalidDroppedForeignKeyName(): void
+    {
+        $table = Table::editor()
+            ->setUnquotedName('t1')
+            ->setColumns(
+                Column::editor()
+                    ->setUnquotedName('c1')
+                    ->setTypeName(Types::INTEGER)
+                    ->create(),
+            )
+            ->create();
+
+        $droppedForeignKeys = ForeignKeyConstraint::editor()
+            ->setUnquotedReferencingColumnNames('c1', 'c2')
```

Comment:
`c2` can be removed from the definition: it's not needed to reproduce the issue and it doesn't exist on the referencing table.